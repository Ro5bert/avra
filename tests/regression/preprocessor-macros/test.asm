.device atmega328p

#define EIGHT (1 << 3)
#define SQR(X) ((X)*(X))

ldi r16, EIGHT
ldi r16, SQR(4)

#define FOOBAR subi
#define IMMED(X) X##i
#define SUBI(X,Y) X ## Y

IMMED(ld) r16, 1
SUBI(FOO, BAR) r16, 1

#define REGN(N) r##N
ldi REGN(16), SQR(4)
