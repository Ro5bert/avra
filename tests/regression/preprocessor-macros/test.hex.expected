:020000020000FC
:0A00000008E000E101E0015000E11A
:00000001FF
